controllers/components/avro_schema.go

```diff
@@ -76,8 +78,29 @@ func (as *AvroSchema) DeleteByVersion(version string) (err error) {
 	return
 }
 
-func (as *AvroSchema) CheckDeviation() (err error) {
-	return //TODO
+func (as *AvroSchema) CheckDeviation() (problem, err error) {
+	schema, err := as.SetSchemaNameNamespace()
+	if err != nil {
+		return nil, errors.Wrap(err, 0)
+	}
+
+	as.registry.Client.ResetCache()
+	srschema, err := as.registry.Client.GetLatestSchema(as.Name())
+	if err != nil {
+		srerr, isSrerr := err.(srclient.Error)
+		if isSrerr && srerr.Code == 40401 {
+			// Error code 40401 – Subject not found
+			// mark as no deviation
```

Comment:
I think this should respond with a problem instead of no deviation. This will handle the case where the schema was deleted by someone/something other than the operator.

Reply:
updated by creating a problem instead of an error